src/Generator/Generators/CLI/CLISources.cs

```diff
@@ -1165,7 +1167,8 @@ public struct ParamMarshal
 
             var effectiveParam = new Parameter(param)
             {
-                QualifiedType = new QualifiedType(paramType)
+                QualifiedType = new QualifiedType(paramType),
+                Namespace = param.Namespace
```

Comment:
The name-space should already be set by the copy ctor. If not, fix there, not here.